lib/src/main/java/graphql/nadel/hints/NadelServiceTypenameShadowingMismatchHint.kt

```diff
@@ -0,0 +1,33 @@
+package graphql.nadel.hints
+
+import graphql.nadel.Service
+import graphql.normalized.ExecutableNormalizedField
+
+interface NadelServiceTypenameShadowingHint {
+    fun isShadowingEnabled(): Boolean
+
+    fun onMismatch(
+        oldDecision: Boolean,
+        newDecision: Boolean,
+        service: Service,
+        objectTypeName: String,
+        field: ExecutableNormalizedField,
+    )
+
+    companion object {
+        internal val default = object : NadelServiceTypenameShadowingHint {
+            override fun isShadowingEnabled(): Boolean {
+                return false
+            }
+
+            override fun onMismatch(
```

Comment:
default implementation is empty, I got lost on how is setting behaviour when it is called on [NadelServiceTypeFilterTransform.kt](https://github.com/atlassian-labs/nadel/pull/637/files#diff-ef65360fa390e6d9ac0bf031a350bd2c8138040cf3664acfc1a9d98da8f1107d)

Reply:
The hint is provided by AGG. So the impl will be logging the mismatch.